pinot-common/src/main/java/org/apache/pinot/common/datablock/ZeroCopyDataBlockSerde.java

```diff
@@ -201,10 +201,10 @@ public DataBlock deserialize(DataBuffer buffer, long offset, DataBlock.Type type
               bufferView(buffer, header._variableSizeDataStart + offset, header._variableSizeDataLength));
         case METADATA: {
           Map<Integer, String> exceptions = deserializeExceptions(stream, header);
+          List<DataBuffer> metadata = deserializeMetadata(buffer, header);
```

Comment:
Will the response contain metadata when it contains exception?

Reply:
Sometimes it does. In cases where a worker sends to more than one downstream worker, only one worker will receive the metadata (aka stats).